### PR TITLE
[Feature](Inverted index) add MATCH_ PHRASE query

### PR DESCRIPTION
**Pull request by zzzxl1993**
# Proposed changes

add match phrase query, usage like this:
```
select * from tbl;

id                       | url
23969194503 | https://api.github.com/users/owais-redhunt
23969194503 | https://api.github.com/users/owais-redhunt
23969194503 | https://avatars.githubusercontent.com/u/107148599?
23969194503 | https://avatars.githubusercontent.com/u/107148599?
23969194503 | https://api.github.com/users/danikaze
23969194503 | https://api.github.com/users/danikaze

select * from tbl where url match_any "users/owais-redhunt";
(If a token is matched after word segmentation, it is returned)

id                       | url
23969194503 | https://api.github.com/users/owais-redhunt
23969194503 | https://api.github.com/users/owais-redhunt
23969194503 | https://api.github.com/users/danikaze
23969194503 | https://api.github.com/users/danikaze

select * from tbl where url match_phrase "users/owais-redhunt";
(After word segmentation matches the same position is returned)

id                       | url
23969194503 | https://api.github.com/users/owais-redhunt
23969194503 | https://api.github.com/users/owais-redhunt
```
## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp [hunk 1]
```diff
@@ -349,6 +349,11 @@ void DorisCompoundDirectory::FSIndexInput::readInternal(uint8_t* b, const int32_
     CND_PRECONDITION(_handle->_reader != nullptr, "file is not open");
     std::lock_guard<doris::Mutex> wlock(*_handle->_shared_lock);
 
+    int64_t position = getFilePointer();
```
Comment (zzzxl1993) [inline be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp:352]: 1. Phrase query file pointer is incorrect when skiplist is used
2. Occurs when two terms intersect

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp [hunk 2]
```diff
@@ -222,101 +223,148 @@ Status FullTextIndexReader::query(OlapReaderStatistics* stats, const std::string
...
+                    break;
+                }
+                case InvertedIndexQueryType::EQUAL_QUERY:
+                case InvertedIndexQueryType::MATCH_ALL_QUERY: {
+                    SCOPED_RAW_TIMER(&stats->inverted_index_query_bitmap_op_timer);
+                    query_match_bitmap &= *term_match_bitmap;
+                    break;
+                }
+                case InvertedIndexQueryType::MATCH_PHRASE_QUERY: {
+                    return Status::Error<ErrorCode::INVERTED_INDEX_NOT_SUPPORTED>();
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:360]: need to fix this

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp [hunk 3]
```diff
@@ -222,101 +223,148 @@ Status FullTextIndexReader::query(OlapReaderStatistics* stats, const std::string
...
+        };
+
         roaring::Roaring query_match_bitmap;
-        bool first = true;
         bool null_bitmap_already_read = false;
-        for (auto token_ws : analyse_result) {
-            std::shared_ptr<roaring::Roaring> term_match_bitmap = nullptr;
+        if (query_type == InvertedIndexQueryType::MATCH_PHRASE_QUERY) {
+            std::wstring str_tokens;
+            auto* phrase_query = new lucene::search::PhraseQuery();
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:278]: if cache hit, phrase_query will not be released, memory leaks.

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**